src/zjs_modules.c

```diff
@@ -33,87 +36,169 @@ struct routine_map {
 static u8_t num_routines = 0;
 struct routine_map svc_routine_map[NUM_SERVICE_ROUTINES];
 
-static const jerryx_module_resolver_t *resolvers[] =
+/*****************************************************************
+*   Real board JavaScript module resolver (ASHELL only currently)
+******************************************************************/
+#ifndef ZJS_LINUX_BUILD
+#ifdef ZJS_ASHELL
+// Eval the JavaScript, and return the module.
+static bool javascript_eval_code(const char *source_buffer, ssize_t size, jerry_value_t *ret_val)
 {
-    &jerryx_module_native_resolver
-};
-
-static ZJS_DECL_FUNC(native_require_handler)
+    (*ret_val) = jerry_eval((jerry_char_t *)source_buffer, size, false);
+    if (jerry_value_has_error_flag(*ret_val)) {
+        ERR_PRINT("[ERR] failed to evaluate JS\n");
+        return false;
+    }
+    return true;
+}
+#endif
+// Find the module on the filestystem
+static bool load_js_module_fs (const jerry_value_t module_name, jerry_value_t *result)
 {
-    // args: module name
-    ZJS_VALIDATE_ARGS(Z_STRING);
-
-    jerry_size_t size = MAX_MODULE_STR_LEN;
-    char module[size];
-    zjs_copy_jstring(argv[0], module, &size);
-    if (!size) {
-        return RANGE_ERROR("argument too long");
+    // Currently searching the filesystem is only supported on ashell
+    #ifdef ZJS_ASHELL
```

Comment:
Preprocessor commands should be at far left column, i.e. remove the four spaces here and below on the #else and #endif.